File: back-end/src/main/java/com/capstone/backend/jpa/Medicine.java
package com.capstone.backend.jpa;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "medicine")
public class Medicine {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Byte id;

    @Column(name = "name", nullable = false)
    private String name;
    
    @Column(name = "price", nullable = false)
    private Double price;

    @Column(name = "brand", nullable = false)
    private String brand;

    @Column(name = "stock", nullable = false)
    private Integer stock;

	public Medicine() {}
	
	public Medicine(Byte id, String name, String brand, Integer stock) {
		this.id = id;
		this.name = name;
		this.brand = brand;
		this.stock = stock;
	}
	
    public Byte getId() {
        return id;
    }

    public void setId(Byte id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
    
	public Double getPrice() {
		return price;
	}
	
	public void setPrice(Double price) {
		this.price = price;
	}

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public Integer getStock() {
        return stock;
    }

    public void setStock(Integer stock) {
        this.stock = stock;
    }

	@Override
	public String toString() {
		return "Medicine [id=" + id + ", name=" + name + ", brand=" + brand + ", stock=" + stock + "]";
	}
   
}

File: front-end/src/app/models/medicine.ts
export class Medicine {
    private id: number;
    private name: string;
    private price: number;
    private brand: string;
    private stock: number;

    constructor(id?: number, name?: string, price?: number, brand?: string, stock?: number) {
        this.id = id || 0;
        this.name = name || '';
        this.price = price || 0;
        this.brand = brand || '';
        this.stock = stock || 0;
    }

    get Id() {
        return this.id;
    }

    set Id(value: number) {
        this.id = value;
    }

    get Name() {
        return this.name;
    }

    set Name(value: string) {
        this.name = value;
    }

    get Price() {
        return this.price;
    }

    set Price(value: number) {
        this.price = value;
    }

    get Brand() {
        return this.brand;
    }

    set Brand(value: string) {
        this.brand = value;
    }

    get Stock() {
        return this.stock;
    }

    set Stock(value: number) {
        this.stock = value;
    }

    toString(): string {
        return `Medicine [id=${this.id}, name=${this.name}, brand=${this.brand}, stock=${this.stock}]`;
    }
}